clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -556,6 +575,7 @@ private void handleListOffsetResponse(TopicPartition topicPartition,
         int bytes = 0;
         int recordsCount = 0;
         PartitionRecords<K, V> parsedRecords = null;
+        boolean emptyResponse = false;
```

Comment:
It's a little weird that errors are treated as non-empty responses. Maybe we can avoid the overloading by explicitly checking for an error code in the `if` check before the `movePartitionsToEnd` call.

It's probably also helpful to add a comment explaining the justification for moving the partition to the back when an error is encountered.